enhancements/host-network-configuration.md

```diff
@@ -0,0 +1,439 @@
+DEVICE=${IFNAME}
+BOOTPROTO=none
+ONBOOT=no
+EOF
+done
+```
+
+Next we update the two ignition configuration files generated by the installer
+to include our network configuration.   This example makes use of
+[filetranspiler](https://github.com/ashcrow/filetranspiler).
+
+```
+mv master.ign master.ign.orig
+mv worker.ign worker.ign.orig
+for i in master worker; do python3 ~/filetranspile --pretty -i ${i}.ign.orig -f ./lab -o ${i}.ign; done
+```
+
+### MachineConfig Resource
+
+Once a cluster is able to reach the Machine Config Server, an additional
+interface is available for managing host configuration: the `MachineConfig`
+resource.  The `MachineConfig` resource can be used to add additional host
+configuration for a single Node or pool of Nodes.  The OpenShift documentation
+provides some examples of using this interface, such as [this example of
+configuring an NTP server for the chrony time
+service](https://docs.openshift.com/container-platform/4.4/installing/install_config/installing-customizing.html#installation-special-config-crony_installing-customizing).
+
+This same interface can be used to manage custom NetworkManager configuration
+files.  As long as the network configuration is not needed during the initial
+bootstrapping of a new Node, this interface provides access to any host network
+configuration that’s doable through NetworkManager.
+
+#### Example
+
+Consider a bare metal cluster where the administrator would like to configure a
+VLAN interface after the cluster is up and running.  A `MachineConfig` resource
+can be used to apply a new NetworkManager configuration file to do this.
+
+First, we must craft a config file to drop into the
+`/etc/NetworkManager/system-connections `directory on each host. For example
+purposes, I used this config file, which configures a VLAN interface with a
+static IP instead of relying on DHCP to be present on this VLAN.
+
+In this case I used a VLAN ID of 20, and the NIC is ens3.
+
+```
+[connection]
+id=vlan-vlan20
+type=vlan
+interface-name=vlan20
+autoconnect=true
+[ipv4]
+method=manual
+addresses=172.5.0.2/24
+gateway=172.5.0.1
+[vlan]
+parent=ens3
+id=20
+```
+
+These contents must then be base64 encoded and placed into a `MachineConfig`
+resource. Here's the resource I used:
+
+```
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 00-ocs-vlan
+spec:
+  config:
+    ignition:
+      version: 2.2.0
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,W2Nvbm5lY3Rpb25dCmlkPXZsYW4tdmxhbjIwCnR5cGU9dmxhbgppbnRlcmZhY2UtbmFtZT12bGFuMjAKYXV0b2Nvbm5lY3Q9dHJ1ZQpbaXB2NF0KbWV0aG9kPW1hbnVhbAphZGRyZXNzZXM9MTcyLjUuMC4yLzI0CmdhdGV3YXk9MTcyLjUuMC4xClt2bGFuXQpwYXJlbnQ9ZW5zMwppZD0yMAo=
+        filesystem: root
+        mode: 0600
+        path: /etc/NetworkManager/system-connections/ocs-vlan.conf
```

Comment:
did this work? I've seen some weird behavior here so I've always had a `.nmconnection` suffix on the files I place in this directory.

Reply:
Well ... it worked when I tried it in July of 2019.  I dug this up from https://github.com/openshift-kni/install-scripts/issues/4#issuecomment-515579812